### ISSUE TITLE
ICA OnChanOpenAck fixes

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#314]
## Summary

- [ ] [matches spec](https://github.com/cosmos/ibc/pull/567/files?short_path=5698356#diff-56983566230cc9f705648cb67111ac9a42e097e0de832ed165b6da08331cb299) (missing version check)
- [ ] test active channel is correctly set

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by colin-axner** [comment]
> We don't actually pass Version into the ack step. We have access to the counter-party version but is it even necessary to check this? Maybe it should be removed from the spec.

Yes it is. The version set on our end is the [counterpartyVersion](https://github.com/cosmos/ibc-go/blob/interchain-accounts/modules/core/04-channel/keeper/handshake.go#L328), thus it must match our required version. Otherwise we may run into issues later if we try to create a new version and existing channels have random version strings

**Comment by seantking** [comment]
Right ok, gotcha. Thanks :+1: 

Agree, let's add it. Secondary thought: if we end up going with the approach whereby we pass the address back as the version string, this is somewhat less necessary I imagine. Or perhaps we pass back a version like `ics27-1-{interchain-account address}`

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
